src/python/WMCore/Services/DBS/DBS3Reader.py

```diff
@@ -724,18 +728,20 @@ def listDatasetLocation(self, datasetName):
             raise DBSReaderError(msg)
 
         if not blocksInfo:  # no data location from dbs
-            return list()
+            return list(locations)
 
         for blockInfo in blocksInfo:
-            locations.update(blockInfo['origin_site_name'])
+            locations.add(blockInfo['origin_site_name'])
```

Comment:
here code is not protected against missing key in a dictionary. I think it will be better to do the following:
```
site = blockInfo.get('origin_site_name', None)
if site:
   locations.add(site)
```

Reply:
I was going to say the line above `if not blocksInfo` should give enough protection, but I think there is a minor improvement that can be done here.